### PR TITLE
Replace no campaigns button with apex donation modal

### PR DESCRIPTION
**Pull request by slants0**
When no campaigns are active, the gift-a-meal button defaults to a link to the GoFundMe page. Instead, the button should now open a the pooled donation modal. Modal should link to an apex donation.

image

Remaining to-do:

1. Create Apex project ID in production database
2. Merge PR with backend changes for apex pathing - https://github.com/sendchinatownlove/ruby/pull/311

Previous PR: https://github.com/sendchinatownlove/sendchinatownlove.github.io/pull/443

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -20,6 +20,10 @@ REACT_APP_SQUARE_LOCATION_ID=E4R1NCMHG7B2Y
 REACT_APP_THINK_CHINATOWN_APPLICATION_ID=sandbox-sq0idb-j_jFtQUzesCshLEjBd-Lxg
 REACT_APP_THINK_CHINATOWN_LOCATION_ID=LP8KMVMD1BFPK
 
+# Square publishable keys for Apex
+REACT_APP_Apex_APPLICATION_ID=sq0idp-ayb-cNC1zt49cYM7MZ6nYQ
+REACT_APP_Apex_LOCATION_ID=LJQ6T0REE7VS6
+
```
Comment (slants0) [inline .env.development:26]: Are there alternate credentials to use for development?

Comment (justintmckibben) [reply .env.development:26]: Yeah you can create them via sendchinatownlove@gmail.com. DM us in #engineering

File: src/components/MerchantsPage/gam/NoCampaignsBox.tsx [hunk 2]
```diff
@@ -5,25 +5,40 @@ import {
...
-          e.preventDefault();
-          window.open('https://www.gofundme.com/f/gift-a-meal', '_blank');
-        }}
-      >
+      <Button className="button--filled" onClick={openModal}>
         {t('gamHome.giftButton')}
       </Button>
+      <Modal
+        sellerId="donations-for-apex"
+        sellerName="Donate to Apex"
```
Comment (justintmckibben) [inline src/components/MerchantsPage/gam/NoCampaignsBox.tsx:39]: nit: The id and name should convert to one another, eg) Shunfa Bakery -> shunfa-bakery

Comment (justintmckibben) [reply src/components/MerchantsPage/gam/NoCampaignsBox.tsx:39]: I'd just call it `apex-for-youth :Apex for Youth`

File: src/components/ModalPayment/ModalCardDetails/ModalCardDetails.tsx [hunk 3]
```diff
@@ -81,6 +84,9 @@ const ModalCardDetails = ({
   if (sellerId === 'light-up-chinatown') {
     sellerId = '';
     projectId = '1';
+  } else if (sellerId === 'donations-to-apex') {
+    sellerId = '';
+    projectId = '2';
```
Comment (justintmckibben) [inline src/components/ModalPayment/ModalCardDetails/ModalCardDetails.tsx:89]: nit: would be nice to have these as a constant map somewhere. Something like selllerIdToProjectId

**Review by justintmckibben — APPROVED** [review]

File: src/components/MerchantsPage/gam/NoCampaignsBox.tsx (hunk 2 above)
Comment (slants0) [reply src/components/MerchantsPage/gam/NoCampaignsBox.tsx:39]: You got it

File: src/components/ModalPayment/ModalCardDetails/ModalCardDetails.tsx [hunk 4]
```diff
@@ -73,14 +73,23 @@ const ModalCardDetails = ({
...
 
-  if (sellerId === 'light-up-chinatown') {
+  const projectIdsMap = {
+    'light-up-chinatown': 1,
+    'apex-for-youth': 2
+  };
+
+  // if a project, map to projectId and remove sellerId
+  if (projectIdsMap[sellerId]) {
+    projectId = projectIdsMap[sellerId];
```
Comment (slants0) [inline src/components/ModalPayment/ModalCardDetails/ModalCardDetails.tsx:91]: @justintmckibben this is what I landed on, is this what you had in mind?

**Comment by justintmckibben** [comment]
yes looks great!

**Review by jacoblb64 — APPROVED** [review]

**Pull request merged**